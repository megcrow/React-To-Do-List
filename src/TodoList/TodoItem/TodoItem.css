.list-item-container {
    margin-bottom: 15px;
    display: flex;
    align-items: center;
}

.list-item {
    display: inline-block;
    color: #333;
    background-color: rgba(255,255,255,.5);
    border-radius: 5px;
    border: solid  rgba(255,255,255,0);
    list-style: none;
    text-align: center;
    padding: 10px;
    word-wrap: break-word;
    width: 165px;
    width: 169px;
    margin-right: 9px;
    font-size: 16px;

}

.list-item::placeholder{
    color:  #333;
    opacity: 1;
    font-size: 16px;
}

.delete-button {
    border-radius: 5px;
    margin: 0;
    height: 37px;
    padding: 0 5px;
    font-size: 13px;
    width: 52px;
}
