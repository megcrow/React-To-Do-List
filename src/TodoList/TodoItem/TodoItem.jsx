import React from 'react';

import './TodoItem.css';

function TodoItem(props) {
    const {
        item,
        deleteItem,
        editItem
    } = props;
    const {
        key,
        text
    } = item;

    function _deleteItem() {
        deleteItem(key);
    }

    function _editItem(e) {
        const {
            value
        } = e.target;

        editItem(key, value);
    }

    return (
        <div className="list-item-container">
            <form onBlur={_editItem}> 
                <textarea
                    className="list-item"  
                    type="text"
                    placeholder={text}
                />
            </form>
            <button
                className="delete-button" 
                onClick={_deleteItem}
            >
                delete
            </button>
        </div> 
    );
};

export default TodoItem;
