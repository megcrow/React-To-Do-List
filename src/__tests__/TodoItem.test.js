import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import Enzyme,{ mount } from 'enzyme';
require('jest-extended');

import TodoItem from '../components/TodoList/TodoItem/TodoItem';


Enzyme.configure({ adapter: new Adapter() })

describe('TodoItem', () => {
    let wrapper;
    let deleteItem = jest.fn();
    let editItem = jest.fn();
    beforeEach(() => {
        wrapper = mount(
            <TodoItem
            item={{key: 1, text: 'mock text'}}
            deleteItem={deleteItem}
            editItem={editItem}
            />
        );
    });
    test('TodoItem component should match snapshot', () => {
        expect('wrapper').toMatchSnapshot();
    });

    test('TodoItem should have form component', () => {
        const form = wrapper.find('form')
        expect(
           form.hasClass('todo-item-form')
        ).toBe(true);
    });

    test('TodoItem should have text area component that displays the todo item text', () => {
        const listItem = wrapper.find('.list-item');
        expect(
            listItem.props()
        ).toContainEntry(['placeholder','mock text']);
    });

    test('TodoItem should have delete button', () => {
        expect(
            wrapper.find('.delete-button').text()
        ).toBe('delete');
    });

    describe('The todo item text area should be edited on blur', () => {
        beforeEach(() => {
            const todoItemForm = wrapper.find('form');
            todoItemForm.simulate('blur');
            wrapper.setProps({item:{key:'2', text:'mock edit'}})
        });

        test('fires blur event', () => {
            expect(editItem).toBeCalled()
        });

        test('should update the state property `text`', () => {
            const listItem = wrapper.find('.list-item');
            expect(
                listItem.props()
            ).toContainEntry(['placeholder','mock edit']);
        })
    });

    test('clicking the delete button should call deleteItem', () => {
        const deleteButton = wrapper.find('.delete-button');
        deleteButton.simulate('click');
        expect(deleteItem).toBeCalled();
    });
})
