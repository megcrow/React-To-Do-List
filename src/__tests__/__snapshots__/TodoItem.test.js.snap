// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoItem TodoItem component should match snapshot 1`] = `"wrapper"`;
